### PR TITLE
[DONT MERGE] Update /null-safety for beta

### PR DESCRIPTION
**Pull request by mit-mit**
Some updates:

- Incorporate the principles we laid our over last few blog posts
- Tone down warnings about use now that it's in beta
- Correct various paths
- Remove steps no longer needed

### CODE REVIEW

**Comment by mit-mit** [comment]
Staged here: https://mit-staging.web.app/null-safety#enable-null-safety

**Review by kwalrath — COMMENTED** [review]
Oops, I thought I'd already submitted these comments.

**Comment by kwalrath** [comment]
We should add a link or two to the existing migration guide

**Comment by mit-mit** [comment]
> We should add a link or two to the existing migration guide

Which Url is that?

**Comment by kwalrath** [comment]
I think I meant https://dart.dev/null-safety/migration-guide. The page already exists, and #2739 will give it real content.

File: src/null-safety/index.md [hunk 1]
```diff
@@ -3,62 +3,56 @@ title: Sound null safety
...
+*  **Non-nullable by default**. Unless you explicitly tell Dart that a variable
+   can be null, it's considered non-nullable. This default was chosen
+   after research found that non-null was by far the most common choice in APIs.
+
+* **Incrementally adoptable**. You choose _what_ to migrate to null safety, and _when_.
+  You can migrate incrementally, mixing null-safe and
+  non-null-safe code in the same project. We provide tools to help you
+  with the migration.
+
+```suggestion
```
Comment (kwalrath) [inline src/null-safety/index.md:39]: Yikes, the suggestion line got left in, somehow! It breaks formatting.

**Comment by mit-mit** [comment]
> I think I meant https://dart.dev/null-safety/migration-guide. 

Added two links

**Comment by mit-mit** [comment]
Staging has been updated: https://mit-staging.web.app/null-safety

**Comment by mit-mit** [comment]
@kwalrath any more feedback on this one, or is it approved?

**Review by kwalrath — APPROVED** [review]

**Comment by mit-mit** [comment]
@kwalrath added one more commit: https://github.com/dart-lang/site-www/pull/2734/commits/5b32909f51bd43449fc132d5aebf09e17dcea138. This is a suggestion Tim made over in the blog post draft.

File: src/null-safety/index.md [hunk 2]
```diff
@@ -311,71 +305,45 @@ int value = aMap['one'] ?? 0;
...
-the Dart [experiment flags documentation][experiment-flags].
-
-{{ site.alert.version-note }}
-  After null safety launches in a beta or stable release,
-  you won't need to enable the null safety experiment.
-{{ site.alert.end }}
+{{site.alert.note}}
+The 2.12 SDK constraint above ends in **`-0`**. This uses
+[semantic versioning notation](https://semver.org/)
+to allow 2.12.0 prereleases, such as the `2.12.0-29.10.beta` beta prerelease.
```
Comment (kwalrath) [inline src/null-safety/index.md:329]:
```suggestion
  The 2.12 SDK constraint above ends in **`-0`**.
  This use of [semantic versioning notation](https://semver.org/)
  allows 2.12.0 prereleases, such as the `2.12.0-29.10.beta` beta prerelease.
```

Comment (kwalrath) [reply src/null-safety/index.md:329]: Looks good! Just have a minor suggestion.

**Comment by mit-mit** [comment]
> We usually indent alert content. Is there a reason we shouldn't?

I don't think it makes any difference, and I thought it was added by mistake here. Happy to add it back.

**Pull request merged**